examples/basic/shared/src/main/scala/sample/algebra/DocumentedApi.scala

```diff
@@ -0,0 +1,40 @@
+package sample.algebra
+
+import io.circe.generic.JsonCodec
+
+trait DocumentedApi
+  extends endpoints.algebra.DocumentedEndpoints
+    with endpoints.algebra.DocumentedOptionalResponses
+    with endpoints.algebra.DocumentedBasicAuthentication
+    with endpoints.algebra.DocumentedJsonEntities {
```

Comment:
This is the definition of the documented endpoints. The idea is that the DSL is the same as for non-documented endpoints, but with additional parameters carrying documentation.